projects/demo-integrations/cypress/tests/kit/time/time-mode.cy.ts

```diff
@@ -345,5 +345,65 @@ describe('Time', () => {
                 });
             });
         });
+
+        describe('HH', () => {
+            describe('default segments', () => {
+                beforeEach(() => {
+                    cy.visit(`/${DemoPath.Time}/API?mode=HH`);
+                    cy.get('#demo-content input')
+                        .should('be.visible')
+                        .first()
+                        .focus()
+                        .clear()
+                        .type('{moveToStart}')
+                        .as('input');
+                });
+
+                it('default segments, 2| => type 5 => 2|', function () {
+                    cy.get('@input')
+                        .type('2')
+                        .should('have.value', '2')
+                        .should('have.prop', 'selectionStart', '2'.length)
+                        .should('have.prop', 'selectionEnd', '2'.length)
+                        .type('5')
+                        .should('have.value', '2')
+                        .should('have.prop', 'selectionStart', '2'.length)
+                        .should('have.prop', 'selectionEnd', '2'.length);
+                });
+
+                it('default segments, 2| => type 3 => 23|', function () {
+                    cy.get('@input')
+                        .type('2')
+                        .should('have.value', '2')
+                        .should('have.prop', 'selectionStart', '2'.length)
+                        .should('have.prop', 'selectionEnd', '2'.length)
+                        .type('3')
+                        .should('have.value', '23')
+                        .should('have.prop', 'selectionStart', '23'.length)
+                        .should('have.prop', 'selectionEnd', '23'.length);
+                });
+            });
+
+            describe('max hours 11', () => {
+                beforeEach(() => {
+                    cy.visit(`/${DemoPath.Time}/API?mode=HH&timeSegmentMaxValues$=1`);
+                    cy.get('#demo-content input')
+                        .should('be.visible')
+                        .first()
+                        .focus()
+                        .clear()
+                        .type('{moveToStart}')
```

Comment:
I think we can drop this line because caret index is already `0` after `.clear()`